lib/linguist/generated.rb

```diff
@@ -311,5 +312,17 @@ def compiled_cython_file?
       return false unless lines.count > 1
       return lines[0].include?("Generated by Cython")
     end
+
+    # Internal: Is this a metadata file from Unity3D?
+    #
+    # Unity3D Meta files start with:
+    #   fileFormatVersion: X
+    #   guid: XXXXXXXXXXXXXXX
+    #
+    # Return true or false
+    def generated_unity3d_meta?
+      return false unless extname == '.meta'
+      return lines[0].include?("fileFormatVersion: ")
```

Comment:
I think you should check for the number of lines: `return false unless lines.count > 1`